openpype/modules/deadline/plugins/publish/submit_publish_job.py

```diff
@@ -448,7 +448,11 @@ def _create_instances_for_aov(self, instance_data, exp_files):
             preview = False
             if app in self.aov_filter.keys():
                 for aov_pattern in self.aov_filter[app]:
-                    if re.match(aov_pattern, aov):
+                    # Matching against the AOV pattern in the render files
+                    # In order to match the AOV name, we must compare against the render filename string
+                    # We are grabbing the render filename string from the collection that we have grabbed from expected files (exp_files)
+                    render_file_name = os.path.basename(col[0])
```

Comment:
According to [these lines](https://github.com/pypeclub/OpenPype/blob/509f3289418a297ac17e3a0de93a987390cd8370/openpype/modules/deadline/plugins/publish/submit_publish_job.py#L403-L408) can `col` also be a single "remainder" which is not a collection but a single filepath? If so, doing `col[0]` would just return the first character of that string.

You can examine a similar "problem" being worked around [here](https://github.com/pypeclub/OpenPype/blob/509f3289418a297ac17e3a0de93a987390cd8370/openpype/modules/deadline/plugins/publish/submit_publish_job.py#L428-L431).

Reply:
Damn. You're right, that should be caught.